### ISSUE TITLE
WARNING Skipping checkpoint save at epoch 4: EMA contains NaN/Inf

### ISSUE THREAD

**Issue by sarpx** [ultralytics/ultralytics#24871]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

İlk defa böyle bir hata ile karşılaşıyorum

### Environment

Ultralytics 8.4.70  Python-3.10.13 torch-2.2.0 CUDA:0 

### Minimal Reproducible Example

WARNING Skipping checkpoint save at epoch 4: EMA contains NaN/Inf

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sarpx, thank you for reporting this issue with Ultralytics 🚀 This is an automated response to help get things moving quickly, and an Ultralytics engineer will also assist you soon.

Since this appears to be a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) that shows the full training setup so the team can investigate more effectively. The warning text alone is usually not enough to isolate the root cause 🔍

Please include as much of the following as possible:
- the exact training command or Python script
- model used
- dataset YAML
- a small sample of images/labels if possible
- training arguments such as `batch`, `imgsz`, `device`, `lr0`, `optimizer`, and `amp`
- complete console log from the start of training until the warning appears
- whether the issue still occurs after upgrading `ultralytics`

We also recommend a quick visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello. This isn't a bug. It means your training experienced NaN loss and the checkpoint wasn't saved because it's corrupted

**Comment by sarpx** [comment]
@Y-T-G  Actually, we were running a new training without changing the training set, and since we install the latest version of YOLO before each training, I thought this might be a mistake. Upon closer inspection, we found two broken maps, but these maps were broken before and didn't cause any errors.

**Comment by Y-T-G** [comment]
Newer Ultralytics versions skip saving NaN checkpoint because the checkpoints are not useable

**Comment by sarpx** [comment]
@Y-T-G , @glenn-jocher  Whatever was done in the recent updates, VRAM usage increased, causing the batch count to drop significantly. This is lowering the quality of training; there wasn't this much VRAM loss before. Whatever was done to increase VRAM usage, it makes me think Yolo is regressing. I think this needs to be investigated.

**Comment by Y-T-G** [comment]
Can you provide more information about your training? Without knowing the model you're training and the exact training command, it's hard to debug.

Lower batch size shouldn't cause deterioration in training because Ultralytics uses gradient accumulation.

**Comment by sarpx** [comment]
@Y-T-G   The `yolo segment train model=yolo26x-seg.pt data=1772108756960.yaml epochs=250 imgsz=640 overlap_mask=False save_period=50 batch=32 amp=True patience=16 exist_ok=True name=1772108756960` configuration worked before. Then we increased the number of maps, updated the `yolo` version, and started training with the new data. We started getting `nan` errors. We fixed the `nan` errors, but the training still ended after 6 periods. Then we trained the previous data, which had given us successful results, with the new version, and it crashed again. So there must be something in the new version that consumes VRAM. Because there's a huge version difference, I don't know exactly when this problem started. I know `yolo` uses `nbsp.p` = 64, and there shouldn't be a problem when the batch count drops to 24, but the tests don't show this. The lower the batch count, the worse the results get.

**Comment by Y-T-G** [comment]
Was the previous training also with YOLO26x?

**Comment by sarpx** [comment]
@Y-T-G  Yes, they were both the same data and model; the only difference was the YOLO version.

**Comment by Y-T-G** [comment]
Which version did you use before?
